File: backend/src/main/java/edu/cit/cabasag/barberconnect/feature/social/Comment.java
package edu.cit.cabasag.barberconnect.feature.social;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

/** Vertical Slice Architecture — Social Feature: Comment on a barber post. */
@Data @NoArgsConstructor @AllArgsConstructor
public class Comment {
    private String comment_id;
    private String post_id;
    private String user_id;
    private String commenterName;
    private String profileImageUrl;
    private String content;
    private Boolean isActive = true;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: mobile/app/src/main/java/edu/cit/cabasag/barberconnect/model/Comment.kt
package edu.cit.cabasag.barberconnect.model

import com.google.gson.annotations.SerializedName

data class Comment(
    @SerializedName("comment_id") val commentId: String? = null,
    @SerializedName("post_id") val postId: String? = null,
    @SerializedName("user_id") val userId: String? = null,
    val content: String? = null,
    val isActive: Boolean? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null,
    val commenterName: String? = null,
    val profileImageUrl: String? = null
)
